iree/compiler/Dialect/HAL/Target/LLVM/internal/UnixLinkerTool.cpp

```diff
@@ -28,7 +28,11 @@ class UnixLinkerTool : public LinkerTool {
     if (!toolPath.empty()) return toolPath;
 
     // No explicit linker specified, search the environment for common tools.
-    toolPath = findToolInEnvironment({"ld", "ld.gold", "ld.lld"});
+    if (useClangDriver()) {
+      toolPath = findToolInEnvironment({"clang"});
+    } else {
+      toolPath = findToolInEnvironment({"ld", "ld.gold", "ld.lld"});
+    }
```

Comment:
Here's another spot where I'd rather at least start with the explicit configuration then consider adding implicit switches.

If this can be achieved by setting the linker to clang explicitly (`-iree-llvm-system-linker-path=` flag, `IREE_LLVMAOT_SYSTEM_LINKER_PATH` env var), then we wouldn't need all this branching in this file.